cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/writer/LineageWriterDatasetFramework.java

```diff
@@ -191,8 +192,10 @@ private void doWriteLineage(Id.DatasetInstance datasetInstanceId, AccessType acc
     Id.Run programRunId = programContext.getRun();
     if (programRunId != null) {
       Id.NamespacedId componentId = programContext.getComponentId();
```

Comment:
should ``ProgramContext.getComponentId()`` be changed to return a ``NamespacedEntityId`` in this PR itself? That way, you wouldn't need the following null check